### PR TITLE
Add hash to PeerChainInfo

### PR DESCRIPTION
**Pull request by mzhangmzz**
This is a preparation for a following PR that will use not just block height but also hash of the highest block from a peer to determine whether the node should enter sync mode

### CODE REVIEW

File: chain/network/src/peer/peer_actor.rs [hunk 1]
```diff
@@ -597,6 +610,25 @@ impl PeerActor {
                                 }
                             }
                         }));
+                        // Send latest block periodically
+                        ctx.spawn(wrap_future({
+                            let conn = conn.clone();
+                            let state = act.network_state.clone();
+                            async move {
+                                let mut interval =
+                                    tokio::time::interval(SYNC_LATEST_BLOCK_INTERVAL.try_into().unwrap());
```
Comment (pompon0) [inline chain/network/src/peer/peer_actor.rs:619]: use time::Interval::new instead. It allows to fake time progress in tests.

**Review by pompon0 — APPROVED** [review]
LG! Please consider adding a test in some followup PR.

**Comment by mzhangmzz** [comment]
> LG! Please consider adding a test in some followup PR.

Do you have any suggestions with how to implement tests for this? Are there some existing tests that I can borrow?

**Comment by pompon0** [comment]
> Yes. Do you have any suggestions with how to implement tests for this? Are there some existing tests that I can borrow?

You can implement the following scenario:
1. spawn a PeerManagerActor
2. set SetChainInfo
3. spawn a new connection
4. in a loop (for example 5 times): advance time, then await a Block message on the connection.

PTAL at the tests in peer_manager/tests, as an example.

**Pull request merged**